packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2128,6 +2137,22 @@ export class SnapController extends BaseController<
     try {
       const processedPermissions =
         this.#processSnapPermissions(initialPermissions);
+
+      const excludedPermissionErrors = Object.keys(processedPermissions).reduce<
+        string[]
+      >((errors, permission) => {
+        if (Object.keys(this.#excludedPermissions).includes(permission)) {
```

Comment:
`hasProperty` might be faster than generating a list of keys and iterating each time?